x/ecocredit/module/module.go

```diff
@@ -126,6 +128,16 @@ func (a Module) ValidateGenesis(cdc codec.JSONCodec, _ sdkclient.TxEncodingConfi
 		return err
 	}
 
+	var genesisMap map[string]json.RawMessage
+	if err := json.Unmarshal(bz, &genesisMap); err != nil {
+		return err
+	}
+
+	basketBalances := []*basketapi.BasketBalance{}
+	if err := json.Unmarshal(genesisMap[proto.MessageName(&basket.BasketBalance{})], &basketBalances); err != nil {
+		return err
+	}
```

Comment:
since this data is already in `bz`, could we do this operation inside of `ValidateGenesis`? otherwise we are shoving extra memory into the `ValidateGenesis` parameters